app-route.html

```diff
@@ -198,8 +184,31 @@
           return;
         }
 
+        // Copy queryParams and track whether there are any differences compared
+        // to the existing query params.
+        var copyOfQueryParams = {};
+        var anythingChanged = false;
+        for (var key in queryParams) {
+          copyOfQueryParams[key] = queryParams[key];
+          if (anythingChanged ||
+              !this.queryParams ||
```

Comment:
Yeah, this loop is doing double duty in copying and checking so we can't break early here.